deps/chakrashim/core/lib/Jsrt/Jsrt.cpp

```diff
@@ -3433,6 +3433,178 @@ CHAKRA_API JsStringFree(_In_ char* stringValue)
     return JsNoError;
 }
 
+
+// -------- Experimental new String APIs ----------------------------------
+
+
+template <class SrcChar, class DstChar>
+static void CastCopy(const SrcChar* src, DstChar* dst, size_t count)
+{
+    const SrcChar* end = src + count;
+    while (src < end)
+    {
+        *dst++ = static_cast<DstChar>(*src++);
```

Comment:
This is to support v8 String's "OneByte" (latin1) API. Yes, in that case the higher 8 bits are discarded (by design). If the chars are really latin1 this is not a problem. If the chars are not, reading OneByte result is wrong anyway.